dome9/resource_dome9_attach_iam_safe_test.go

```diff
@@ -0,0 +1,128 @@
+package dome9
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+
+	"github.com/dome9/dome9-sdk-go/services/cloudaccounts"
+	"github.com/dome9/dome9-sdk-go/services/cloudaccounts/aws"
+	"github.com/terraform-providers/terraform-provider-dome9/dome9/common/resourcetype"
+	"github.com/terraform-providers/terraform-provider-dome9/dome9/common/testing/environmentvariable"
+	"github.com/terraform-providers/terraform-provider-dome9/dome9/common/testing/method"
+	"github.com/terraform-providers/terraform-provider-dome9/dome9/common/testing/variable"
+)
+
+func TestAccResourceAttachIAMSafeBasic(t *testing.T) {
+	var cloudAccountResponse aws.CloudAccountResponse
+	resourceTypeAndName, _, generatedName := method.GenerateRandomSourcesTypeAndName(resourcetype.AttachIAMSafeToAwsCloudAccount)
+	awsTypeAndName, _, awsGeneratedName := method.GenerateRandomSourcesTypeAndName(resourcetype.CloudAccountAWS)
+	awsCloudAccountHCL := getCloudAccountAWSResourceHCL(awsGeneratedName, variable.CloudAccountAWSOriginalAccountName, os.Getenv(environmentvariable.CloudAccountAWSEnvVarArn), "")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccCloudAccountAWSEnvVarsPreCheck(t)
+			testAccAttachIAMSafeEnvVarsPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAttachIAMSafeDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAttachIAMSafeConfigure(awsCloudAccountHCL, awsTypeAndName, resourceTypeAndName, generatedName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAttachIAMSafeExists(resourceTypeAndName, &cloudAccountResponse),
+					resource.TestCheckResourceAttr(resourceTypeAndName, "aws_group_arn", os.Getenv(environmentvariable.AttachIAMSafeEnvVarGroupArn)),
+					resource.TestCheckResourceAttr(resourceTypeAndName, "aws_policy_arn", os.Getenv(environmentvariable.AttachIAMSafeEnvVarPolicyArn)),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAttachIAMSafeDestroy(s *terraform.State) error {
+	apiClient := testAccProvider.Meta().(*Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != resourcetype.AttachIAMSafeToAwsCloudAccount {
+			continue
+		}
+
+		awsCloudAccountResponse, _, err := apiClient.cloudaccountAWS.Get(cloudaccounts.QueryParameters{ID: rs.Primary.ID})
+
+		if err == nil {
+			return fmt.Errorf("id %s already exists", rs.Primary.ID)
+		}
+
+		if awsCloudAccountResponse != nil {
```

Comment:
don't you need to check if `awsCloudAccountResponse.IamSafe` is nil? The destroy process for the attachment don't delete the entire cloud account, just the `iam-safe` part